File: Problems/43105/Problem.js
function solution(triangle) {
    var answer = 0;
    
    answer = triangle.reduceRight((acc, cur) => {
        let arr = [];
        acc.reduce( (a, c) => {
            arr.push( Math.max(a, c) );
            return c;
        });
        //console.log(arr);
        var sumArr = arr.map( (num, idx) => num + cur[idx] );
        //console.log(sumArr);
        return sumArr;
    });
    
    return answer[0];
}

console.log(solution([[7], [3, 8], [8, 1, 0], [2, 7, 4, 4], [4, 5, 2, 6, 5]]));

File: Problems/43105/Problem.py
from functools import reduce

maxArr = []

def reduceCol(acc, cur):
    maxArr.append(max(acc, cur))
    return cur

def reduceRow(acc, cur):
    #print(acc),
    #print(cur),
    reduce(reduceCol, acc)
    #print(maxArr)
    newAcc = [x + y for x, y in zip(maxArr, cur)]
    #print(newAcc)
    del maxArr[:]
    return newAcc

def solution(triangle):
    triangle = list(reversed(triangle))
    return reduce(reduceRow, triangle)[0]

print(solution([[7], [3, 8], [8, 1, 0], [2, 7, 4, 4], [4, 5, 2, 6, 5]]))